File: trunk/PYTHON/pyzbus/test/rpc/rpc_worker.py
# -*- coding: utf-8 -*-
from pyzbus import WorkerPool, Remote, JsonServiceHandler

class MyService(object): 
    
    @Remote()
    def echo(self, ping):
        return ping
    
    @Remote('user')
    def user(self, username):
        return {'Name': username, 'Addr': u'中文'}
    
    @Remote()
    def plus(self, a, b):
        return a + b 

handler = JsonServiceHandler(MyService())
pool = WorkerPool(service = 'rpc')
pool.run(2, handler)

File: trunk/MS.NET/clrzbus/examples/rpc/RpcWorker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus;
namespace zbus
{
    class User
    {
        public string Name;
        public string Addr;
    }
    class MyService
    {
        [Remote]
        public string echo(string msg)
        {
            return msg;
        }

        [Remote]
        public int plus(int a, int b)
        {
            return a + b;
        }

        [Remote]
        public User user(string name)
        { 
            User user = new User();
            user.Name = name;
            user.Addr = "深圳";
            return user;
        }

    }
    class RpcWorker
    {
        public static void Main(string[] args)
        {
            WorkerPoolConfig config = new WorkerPoolConfig();
            config.Service = "rpc";

            WorkerPool pool = new WorkerPool(config);
            ServiceHandler handler = new JsonServiceHandler(new MyService());
            pool.Run(2, handler); //run with 2 threads
        } 
    }
}
